services/user-tenant-service/src/main/java/com/sourcefuse/jarc/services/usertenantservice/controller/TenantUserController.java

```diff
@@ -0,0 +1,263 @@
+    @Valid @RequestBody UserDto userDto,
+    @PathVariable("id") UUID id
+  ) {
+    if (id == null) {
+      throw new ResponseStatusException(
+        HttpStatus.BAD_REQUEST,
+        "${tenant.id.not.specified}"
+      );
+    }
+
+    userDto.setTenantId(id);
+    userDto
+      .getUserDetails()
+      .setEmail(userDto.getUserDetails().getEmail().toLowerCase());
+    userDto
+      .getUserDetails()
+      .setUsername(userDto.getUserDetails().getUsername().toLowerCase());
+    /** map created for ruling with options */
+    Map<String, String> option = new HashMap<>();
+    option.put("authId", userDto.getAuthId());
+    option.put("authProvider", userDto.getAuthProvider());
+    IAuthUserWithPermissions currentUser =
+      (IAuthUserWithPermissions) SecurityContextHolder
+        .getContext()
+        .getAuthentication()
+        .getPrincipal();
+    //tnUsrService.create(userDto, currentUser, option);
+
+    return new ResponseEntity<>("", HttpStatus.CREATED);
+  }
+
+  @GetMapping("")
+  public ResponseEntity<Object> gtUsrTntByRole(@PathVariable("id") UUID id) {
+    IAuthUserWithPermissions currentUser = getiAuthUserWithPermissions(id);
+    Predicate predicate = null;
+    Map<String, Object> map = new HashMap<>();
+    if (
+      currentUser
+        .getPermissions()
+        .contains(PermissionKey.VIEW_TENANT_USER_RESTRICTED.toString()) &&
+      currentUser.getTenantId() == id
+    ) {
+      //      map =
+      //        tnUsrService.checkViewTenantRestrictedPermissions(
+      //          currentUser,
+      //          predicate,
+      //          UserView.class
+      //        );
+      predicate = (Predicate) map.get(CommonConstants.PREDICATE);
+    }
+    CriteriaBuilder cb = map.get(CommonConstants.BUILDER) != null
+      ? (CriteriaBuilder) map.get(CommonConstants.BUILDER)
+      : em.getCriteriaBuilder();
+    CriteriaQuery<UserView> cq = map.get(CommonConstants.CRITERIA_QUERY) != null
+      ? (CriteriaQuery) map.get(CommonConstants.CRITERIA_QUERY)
+      : cb.createQuery(UserView.class);
+    Root<UserView> root = cq.from(UserView.class);
+
+    if (predicate != null) {
+      predicate =
+        cb.and(predicate, cb.equal(root.get(CommonConstants.TENANT_ID), id));
+    } else {
+      predicate = cb.equal(root.get(CommonConstants.TENANT_ID), id);
+    }
+    cq.where(
+      predicate,
+      cb.notEqual(root.get("roleType"), CommonConstants.SUPER_ADMIN_ROLE_TYPE)
+    );
+    //doubt return UserDto
+    return new ResponseEntity<>(
+      new UserView()/*tnUsrService.getUserView(cq)*/,
+      HttpStatus.OK
+    );
+  }
+
+  @GetMapping("/view-all")
+  public ResponseEntity<Object> findAllUsers(@PathVariable("id") UUID id) {
+    List<UserView> usrList = tnUsrService.getAllUsers(id, UserView.class);
+    //nonRestrictedUserViewRepo ::doubt
+    return new ResponseEntity<>(usrList, HttpStatus.OK);
+  }
+
+  @GetMapping("/count")
+  public ResponseEntity<Object> cntUsrTenant(@PathVariable("id") UUID id) {
```

Comment:
function name is not readable
 ```suggestion
   public ResponseEntity<Object> cntUsrTenant(@PathVariable("id") UUID id) {
   public ResponseEntity<Object> userTenantCount(@PathVariable("id") UUID id) {
 ```

Reply:
already done